client/src/components/squids/SquidForm.jsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-to-interactive-role */
+import React, { useState } from "react"
+
+import { useForm } from "react-hook-form"
+import { useQueryClient, useMutation } from "react-query"
+
+import "../../style/form.pcss"
+import { ApiClient } from "../../backend/ApiClient"
+
+export const SquidForm = () => {
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm()
+  const [submitMessage, setSubmitMessage] = useState("")
+
+  const resetForm = () => {
+    reset({
+      name: "",
+      species: "",
+      specialPower: "",
+      experiencePoints: "",
+    })
+    setSubmitMessage("")
+  }
+
+  const addSquidMutation = useMutation((formPayload) => {
+    ApiClient.post("/squids", formPayload)
+  })
+
+  const queryClient = useQueryClient()
+
+  const postSquid = (data) => {
+    addSquidMutation.mutate(data, {
+      onSuccess: () => {
+        setSubmitMessage("Squid Added Successfully")
+      },
+      onError: (error) => {
+        // eslint-disable-next-line no-console
+        console.log(error.response.data.message)
+      },
+      onSettled: () => queryClient.invalidateQueries("squids"),
+    })
+  }
+
+  const [showForm, setShowForm] = useState("squids-form__hide")
+  const toggleShowForm = () => {
+    if (showForm === "squids-form__hide") {
+      setShowForm("squid-form__show")
+    } else {
+      setShowForm("squids-form__hide")
+    }
+  }
+
+  let submitButtonText = "Add Squid"
```

Comment:
awesome! this is really great in that it anticipates that there will be a network delay between submitting and receiving an HTTP response